soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -4059,6 +4181,46 @@ void DrawRandoEditor(bool& open) {
                     SohImGui::EnhancementCombobox("gRandomizeShuffleDungeonReward", randoShuffleDungeonRewards, 4, 0);
                     PaddedSeparator();
 
+                    // Keysanity
+                    ImGui::Text(Settings::Keysanity.GetName().c_str());
+                    InsertHelpHoverText(
+                        "Own dungeon - Small Keys can only appear in their respective dungeon.\n"
+                        "\n"
+                        "Any dungeon - Small Keys can only appear inside of any dungon.\n"
+                        "\n"
+                        "Overworld - Small Keys can only appear outside of dungeons.\n"
+                        "\n"
+                        "Anywhere - Small Keys can appear anywhere in the world."
+                    );
+                    SohImGui::EnhancementCombobox("gRandomizeKeysanity", randoShuffleSmallKeys, 6, 1);
+                    PaddedSeparator();
+
+                    // Gerudo Keys
+                    ImGui::Text(Settings::GerudoKeys.GetName().c_str());
+                    InsertHelpHoverText(
+                        "Any dungeon - Small Keys can only appear inside of any dungon.\n"
+                        "\n"
+                        "Overworld - Small Keys can only appear outside of dungeons.\n"
+                        "\n"
+                        "Anywhere - Small Keys can appear anywhere in the world."
+                    );
+                    SohImGui::EnhancementCombobox("gRandomizeGerudoKeys", randoShuffleGerudoFortressKeys, 4, 0);
```

Comment:
looks like the hover text hasn't been updated to be for the GF keys here

also, have you tested out how this works with open/fast fortress? do only some of the keys get shuffled?

Reply:
I'll test this tonight 👍